pl_bolts/datasets/grasp_and_lift_eeg_dataset.py

```diff
@@ -0,0 +1,238 @@
+    and approximately 30 trials within each series. The number of trials
+    varies for each series. The training set contains the first 8 series
+    for each subject. The test set contains the 9th and 10th series.
+
+    Videos of trials being performed:
+    https://static-content.springer.com/esm/art%3A10.1038%2Fsdata.2014.47/MediaObjects/41597_2014_BFsdata201447_MOESM69_ESM.avi
+    https://static-content.springer.com/esm/art%3A10.1038%2Fsdata.2014.47/MediaObjects/41597_2014_BFsdata201447_MOESM70_ESM.avi
+    (Note: these videos are also available in the S3 bucket)
+
+    Args:
+        root: Path to directory containing train/ and test/ folders
+
+        train: If true, use the train/ directory and load class labels.
+            If false, load the test/ directory (which lacks labels)
+
+        download: If true, download the data if it is not present locally
+
+        num_samples: The number of samples in the returned examples.
+            If None, the the dataset yields the full length trials.
+
+        last_label_only: If true, return only the last sample's labels.
+            Must be used with num_samples.
+
+    Examples:
+        >>> dataset = GraspAndLiftEEGDataset('/data', train=True, download=True, num_samples=1024)
+        Downloading from https://grasplifteeg.nyc3.digitaloceanspaces.com/grasp-and-lift-eeg-detection.zip
+        Downloaded in 283 seconds
+        Extracting /data/grasp-and-lift-eeg-detection.zip to /data
+        Unzipped in 36 seconds
+        >>> len(dataset)
+        17887546
+        >>> data, label = dataset[0]
+        >>> data.shape
+        torch.Size([32, 1024])
+        >>> label.shape
+        torch.Size([6, 1024])
+
+    Labels:
+        1. HandStart
+        2. FirstDigitTouch
+        3. BothStartLoadPhase
+        4. LiftOff
+        5. Replace
+        6. BothReleased
+
+    Reference:
+        Luciw, M., Jarocka, E. & Edin, B. Multi-channel EEG recordings during 3,936
+            grasp and lift trials with varying weight and friction. Sci Data 1, 140047
+            (2014). https://doi.org/10.1038/sdata.2014.47
+
+        https://www.kaggle.com/c/grasp-and-lift-eeg-detection
+
+    License:
+        Original data has been made available under the terms of Attribution 4.0
+        International Creative Commons License (http://creativecommons.org/licenses/by/4.0/).
+    """
+
+    GRASPLIFT_DATA_HEADER = 'id,Fp1,Fp2,F7,F3,Fz,F4,F8,FC5,FC1,FC2,FC6,T7,C3,Cz,C4,T8,TP9,CP5,CP1,CP2,CP6,' + \
+        'TP10,P7,P3,Pz,P4,P8,PO9,O1,Oz,O2,PO10\n'
+
+    GRASPLIFT_EVENTS_HEADER = 'id,HandStart,FirstDigitTouch,BothStartLoadPhase,LiftOff,Replace,BothReleased\n'
+
+    ZIP_URL = 'https://grasplifteeg.nyc3.digitaloceanspaces.com/grasp-and-lift-eeg-detection.zip'
+
+    ZIP_SIZE_BYTES = 980887394
+
+    def __init__(self,
+                 root: str,
+                 train: bool = True,
+                 download: bool = True,
+                 num_samples: int = None,
```

Comment:
Please add `from typing import Optional` at the top of this file.
```suggestion
                 num_samples: Optional[int] = None,
```